documentation/src/main/asciidoc/user_guide/integrations_hibernate_cache.adoc

```diff
@@ -0,0 +1,506 @@
+
+WARNING: It is not possible to specify the configuration this way in WildFly.
+Cache configuration changes in Wildfly should be done either modifying the cache configurations inside the application server configuration, or creating new caches with the desired tweaks and plugging them accordingly.
+See examples below on how entity/collection specific configurations can be applied.
+
+[[caching-provider-infinispan-config-example]]
+.Use custom Infinispan configuration
+====
+[source, XML, indent=0]
+<property
+    name="hibernate.cache.infinispan.cfg"
+    value="my-infinispan-configuration.xml" />
+====
+
+NOTE: If the cache is configured as transactional, Infinispan cache provider automatically sets transaction manager so that the TM used by Infinispan is the same as TM used by Hibernate.
+
+Cache configuration can differ for each type of data stored in the cache.
+In order to override the cache configuration template, use property `hibernate.cache.infinispan._data-type_.cfg` where `_data-type_` can be one of:
+
+* `entity`:
+Entities indexed by `@Id` or `@EmbeddedId` attribute.
+* `immutable-entity`:
+Entities tagged with `@Immutable` annotation or set as `mutable=false` in mapping file.
+* `naturalid`:
+Entities indexed by their `@NaturalId` attribute.
+* `collection`:
+All collections.
+* `timestamps`:
+Mapping _entity type_ -> _last modification timestamp_.
+Used for query caching.
+* `query`:
+Mapping _query_ -> _query result_.
+* `pending-puts`:
+Auxiliary caches for regions using invalidation mode caches.
+
+For specifying cache template for specific region, use region name instead of the `_data-type_`:
+
+[[caching-provider-infinispan-config-cache-example]]
+.Use custom cache template
+====
+[source, XML, indent=0]
+<property
+    name="hibernate.cache.infinispan.entities.cfg"
+    value="custom-entities" />
+<property
+    name="hibernate.cache.infinispan.query.cfg"
+    value="custom-query-cache" />
+<property
+    name="hibernate.cache.infinispan.com.example.MyEntity.cfg"
+    value="my-entities" />
+<property
+    name="hibernate.cache.infinispan.com.example.MyEntity.someCollection.cfg"
+    value="my-entities-some-collection" />
+====
+
+.Use custom cache template in Wildfly
+When applying entity/collection level changes inside JPA applications deployed in Wildfly, it is necessary to specify deployment name and persistence unit name (separated by `#` character):
+
+====
+[source, XML, indent=0]
+<property
+    name="hibernate.cache.infinispan._war_or_ear_name_#_unit_name_.com.example.MyEntity.cfg"
+    value="my-entities" />
+<property
+    name="hibernate.cache.infinispan._war_or_ear_name_#_unit_name_.com.example.MyEntity.someCollection.cfg"
+    value="my-entities-some-collection" />
+====
+
+IMPORTANT: Cache configurations are used only as a template for the cache created for given region.
```

Comment:
Should we warn that in EAP the configuration options (for supportability) are limited? I wouldn't be surprised if users used upstream docs for product.

Reply:
Which options are limited?

Reply:
Well users shouldn't mess with sync/async modes, state transfer, replication timeouts... Maybe @scottmarlow could add more details but I recall someone saying that users should stick with provided configs and change just eviction limits and expiration settings.

Reply:
@pferraro anything to add to ^

Reply:
I see what you mean, I'm adding a warning message.